data_processing.ipynb

```diff
@@ -748,6 +759,421 @@
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>490</th>\n",
+       "      <td>[0.02423272928835719, 0.02342685552305181, 0.0...</td>\n",
+       "      <td>2</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>28</th>\n",
+       "      <td>[0.015755475703704988, 0.01818003315805917, 0....</td>\n",
+       "      <td>3</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>4</th>\n",
+       "      <td>[0.02230706281282789, 0.021776700824309162, 0....</td>\n",
+       "      <td>0</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "<p>153 rows × 2 columns</p>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "                                                    fv label\n",
+       "51   [0.014029436354815213, 0.014280474969277254, 0...     0\n",
+       "370  [0.03026395156144915, 0.030441104676146632, 0....     1\n",
+       "456  [0.018051205585293762, 0.020033789914877715, 0...     2\n",
+       "102  [0.025893012259715704, 0.021900611378626946, 0...     0\n",
+       "36   [0.019407836535601047, 0.0187561280211469, 0.0...     0\n",
+       "..                                                 ...   ...\n",
+       "8    [0.02746254970056796, 0.022406735048318086, 0....     0\n",
+       "107  [0.02350585332112601, 0.025473906998611087, 0....     2\n",
+       "490  [0.02423272928835719, 0.02342685552305181, 0.0...     2\n",
+       "28   [0.015755475703704988, 0.01818003315805917, 0....     3\n",
+       "4    [0.02230706281282789, 0.021776700824309162, 0....     0\n",
+       "\n",
+       "[153 rows x 2 columns]"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "3    53\n",
+      "0    37\n",
+      "2    33\n",
+      "1    30\n",
+      "Name: label, dtype: int64\n"
+     ]
+    }
+   ],
+   "source": [
+    "def split_GPAC_data():\n",
+    "    \"\"\"\n",
+    "    Function that takes all of the data, filters out the rows w/ a label of -1, and then splits \n",
+    "    the data into Training, Validation, and Test \n",
+    "    \"\"\"\n",
+    "    \n",
+    "    all_data = pd.DataFrame(data=None, columns=['fv', 'label'])\n",
+    "    for dataset in gpac_image_data:\n",
+    "        \n",
+    "        # Isolate the image data portion of each dataset\n",
+    "        sub_data = gpac_image_data[dataset]['data']\n",
+    "        sub_data = sub_data.drop(['window_start'], axis=1)\n",
+    "        \n",
+    "        # Now we need to filter out the rows where the 'label' value is -1\n",
+    "        df1 = sub_data[sub_data['label'] > -1]\n",
+    "        all_data = all_data.append(df1)\n",
+    " \n",
+    "    # Now is the splitting of the dataframe into train, test, and validation datafames\n",
+    "    train_data, valid_data, test_data = np.split(all_data.sample(frac=1, random_state=42),\n",
```

Comment:
Should we define a constant at the top of the file to use in random_state, or is just putting 42 here fine?

Reply:
I think here is fine, cause if you wanna rerun it w/ a diff random_state you just run 1 cell. Up to you tho.